oss-internship-2020/libtiff/example/main_sandboxed.cc

```diff
@@ -0,0 +1,305 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <array>
+#include <cstdint>
+#include <cstdlib>
+#include <cstring>
+
+#include "../sandboxed.h"
+#include "sandboxed_api/sandbox2/util/fileops.h"
+#include "sandboxed_api/sandbox2/util/path.h"
+#include "sandboxed_api/vars.h"
+#include "tiffio.h"  // NOLINT(build/include)
+
+// sapi functions:
+//  TIFFTileSize
+//  TIFFOpen
+//  TIFFReadEncodedTile
+//  TIFFSetField
+//  TIFFClose
+//  TIFFReadRGBATile
+//  TIFFGetField
+
+constexpr std::array<uint8_t, 6> kCluster0 = {0, 0, 2, 0, 138, 139};
+constexpr std::array<uint8_t, 6> kCluster64 = {0, 0, 9, 6, 134, 119};
+constexpr std::array<uint8_t, 6> kCluster128 = {44, 40, 63, 59, 230, 95};
+
+namespace {
+
+absl::Status CheckCluster(int cluster, const sapi::v::Array<uint8_t>& buffer,
+                          const std::array<uint8_t, 6>& expected_cluster) {
+  uint8_t* target = buffer.GetData() + cluster * 6;
+
+  if (!std::memcmp(target, expected_cluster.data(), 6)) {
+    return absl::OkStatus();
+  }
+
+  return absl::InternalError(
+      absl::StrCat("Cluster ", cluster, " did not match expected results.\n",
```

Comment:
Not sure what this whole cluster thing is about and why the message here indexes the array in a non linear order (0 1 4 5 2 3 instead of 0 1 2 3 4 5) - maybe adding a comment here could be useful.